### PR TITLE
115210: Omit excess persons from contractor rep

### PR DESCRIPTION
**Pull request by trondlindanger**

### CODE REVIEW

**Comment by jsoi-eq** [comment]
You must add AB#123456 in description to connect PR to PBI.

File: src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitations/GetInvitationsQueryHandler.cs [hunk 1]
```diff
@@ -56,21 +60,39 @@ public async Task<Result<InvitationsResult>> Handle(GetInvitationsQuery request,
             var invitationIds = invitationForQueryDtos.Select(i => i.Id).ToList();
 
             var invitationsWithIncludes = await GetInvitationsWithIncludesAsync(_context, invitationIds, cancellationToken);
+            var codes = invitationsWithIncludes.SelectMany(x => x.Participants).Where(y => y.FunctionalRoleCode != null).Select(y => y.FunctionalRoleCode).Distinct().ToList();
```
Comment (jsoi-eq) [inline src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitations/GetInvitationsQueryHandler.cs:63]: Splitting up linq query up one line per . would increase readability.
I.e.
            var codes = invitationsWithIncludes
                .SelectMany(x => x.Participants)
                .Where(y => y.FunctionalRoleCode != null)
                .Select(y => y.FunctionalRoleCode)
                .Distinct()
                .ToList();

Comment (jsoi-eq) [inline src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitations/GetInvitationsQueryHandler.cs:63]: Suggestion: Can replace .Where(y => y.FunctionalRoleCode != null) with .Where(y => y.HasRole)  for better readability.

Comment (jsoi-eq) [inline src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitations/GetInvitationsQueryHandler.cs:63]: "var codes" - I guess we are talking about functionalRoleCodes?
If so, I would be much obliged if the variable name is renamed to functionalRoleCodes to tax my brain less :)

File: src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetInvitationsQueries/GetInvitations/GetInvitationsQueryHandlerTests.cs [hunk 2]
```diff
@@ -829,13 +887,31 @@ public async Task Handler_ShouldSeparateOutAdditionalConstructionCompanyParticip
             }
         }
 
+        [TestMethod]
+        public async Task Handler_ShouldNotShowParticipants_WhenUsingPersonalEmail()
```
Comment (jsoi-eq) [inline src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetInvitationsQueries/GetInvitations/GetInvitationsQueryHandlerTests.cs:891]: I'm having a bit of trouble verifying that this test is testing what the method name says.

"Should not show participants" -> it seems you are testing AdditionalConstructionCompanyReps for being 0, what is the relationship between participants and AdditionalConstructionCompanyReps?

"WhenUsingPersonalEmail" -> where do you set up that personal email is used in the test?

**Comment by trondkjetillindanger** [comment]
AB#115210

File: src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetInvitationsQueries/GetInvitations/GetInvitationsQueryHandlerTests.cs (hunk 2 above)
Comment (trondlindanger) [reply src/tests/Equinor.ProCoSys.IPO.Query.Tests/GetInvitationsQueries/GetInvitations/GetInvitationsQueryHandlerTests.cs:891]: @jsoi-eq 
See [](https://statoildeveloper.visualstudio.com/Statoil.NextGenProCoSys/_workitems/edit/110737) for details about where/ how the use personal email flag is set. In this test, the flag is set in the method SetupNewDatabase.
"no names should be presented in search results for roles where user personal email is set to true".
 In this test AdditionalConstructionCompanyReps is used as an example to prove the logic. Relationship between participants and AdditionalConstructionCompanyReps is one to one. It is a filtered representation of participants. As defined by GetAdditionalConstructionCompanyReps.

**Review by jsoi-eq — APPROVED** [review]

**Pull request merged**